iothub/device/src/Common/Singleton.cs

```diff
@@ -39,7 +40,7 @@ public Task OpenAsync(TimeSpan timeout, CancellationToken cancellationToken)
             return GetOrCreateAsync(timeout, cancellationToken);
         }
 
-        public Task CloseAsync(CancellationToken cancellationToken)
+        public Task CloseAsync()
```

Comment:
Q - why not throw if cancellation was requested? Does the upstream call not have a cancellation token available?

Reply:
One thing I thought about here was we only Dispose the object and return a task complete. If we cancel this should it be after the dispose? If so do we really want to throw an exception if we're disposing the object?

Reply:
You are correct, throwing an exception after disposing makes no sense.
My idea is to to check for cancellation before performing any processing operation; that way, we do not dispose if a cancellation has been requested.

Reply:
Upon further review this is not used by anyone and it's not public so it's gone.